ms/MSOper/MSMetaData.cc

```diff
@@ -3546,6 +3553,345 @@ void MSMetaData::_computeScanAndSubScanProperties(
     }
 }
 
+#ifdef _OPENMP
+void MSMetaData::_computeScanAndSubScanPropertiesParallel(
+    SHARED_PTR<std::map<ScanKey, MSMetaData::ScanProperties> >& scanProps,
+    SHARED_PTR<std::map<SubScanKey, MSMetaData::SubScanProperties> >& subScanProps,
+    Bool showProgress
+) const {
+    uInt nrows = nRows();
+    SHARED_PTR<Vector<Int> > scans = _getScans();
+    SHARED_PTR<Vector<Int> > fields = _getFieldIDs();
+    SHARED_PTR<Vector<Int> > ddIDs = _getDataDescIDs();
+    SHARED_PTR<Vector<Int> > states = _getStateIDs();
+    SHARED_PTR<Vector<Double> > times = _getTimes();
+    SHARED_PTR<Vector<Int> > arrays = _getArrayIDs();
+    SHARED_PTR<Vector<Int> > observations = _getObservationIDs();
+    SHARED_PTR<Vector<Int> > ant1, ant2;
+    SHARED_PTR<QVD> exposureTimes = _getExposureTimes();
+    SHARED_PTR<QVD> intervalTimes = _getIntervals();
+    vector<uInt> ddIDToSpw = getDataDescIDToSpwMap();
+    _getAntennas(ant1, ant2);
+    uInt nthreads = min((uInt)1000, nrows);
+    uInt chunkSize = nrows/nthreads;
+    if (nrows % nthreads > 0) {
+        ++chunkSize;
+    }
+    _pmCount = 0;
+    SHARED_PTR<ProgressMeter> pm;
+    if (showProgress) {
+        LogIO log;
+        log << LogOrigin("MSMetaDataThreaded", __func__, WHERE)
+            << LogIO::NORMAL << "Compute subscan properties " << LogIO::POST;
+        pm.reset(new ProgressMeter(0, nrows, "Compute subscan info"));
+    }
+    pair<map<ScanKey, ScanProperties>, map<SubScanKey, SubScanProperties> > *fut =
+        new pair<map<ScanKey, ScanProperties>, map<SubScanKey, SubScanProperties> >[nthreads];
+#pragma omp parallel for
+    for (uInt i=0; i<nthreads; ++i) {
+        fut[i] = _getChunkSubScanProperties(
+            scans, fields, ddIDs, states, times, arrays,
+            observations, ant1, ant2, exposureTimes,
+            intervalTimes,  ddIDToSpw,  i * chunkSize,
+            std::min((i + 1) * chunkSize - 1, nrows - 1)
+        );
+#pragma omp critical(progresslock)
+        {
+            if (pm) {
+#pragma omp flush(_pmCount)
```

Comment:
actually keeping it shared is probably simplest, but it should be updated here inside the loop based on the processed chunksize instead of for each single row in another function.